src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/LoadedDocValues.java

```diff
@@ -217,7 +397,6 @@ public void setDocId(int docID) throws IOException {
           values.add(decoder.apply(docValues.nextValue()));
         }
       }
-      values.trimToSize();
```

Comment:
Can this result in higher heap usage? Or is the array already right-sized?

Reply:
Trimming resizes the internal array to match the current number of elements. This will result in more arrays being created and data copy operations. I think it is best to avoid this. It may hold a little more active memory at times, but no more than the size of the largest array.